src/protocols/bdx/BDXMessageUtils.cpp

```diff
@@ -185,22 +158,35 @@ CHIP_ERROR TransferInit::Parse(System::PacketBuffer * aBuffer, TransferInit & aP
 
     SuccessOrExit(bufReader.Read16(&aParsedMessage.mFileDesLength).StatusCode());
 
-    VerifyOrExit(bufReader.OctetsRead() + aParsedMessage.mFileDesLength <= aBuffer->DataLength(),
+    VerifyOrExit(bufReader.OctetsRead() + aParsedMessage.mFileDesLength <= aBuffer.DataLength(),
                  err = CHIP_ERROR_MESSAGE_INCOMPLETE);
     aParsedMessage.mFileDesignator = &bufStart[bufReader.OctetsRead()];
 
     // Rest of message is metadata (could be empty)
-    if (bufReader.OctetsRead() + aParsedMessage.mFileDesLength < aBuffer->DataLength())
+    if (bufReader.OctetsRead() + aParsedMessage.mFileDesLength < aBuffer.DataLength())
     {
-        uint16_t metadataStartIndex    = (uint16_t)(bufReader.OctetsRead() + aParsedMessage.mFileDesLength);
+        uint16_t metadataStartIndex    = static_cast<uint16_t>(bufReader.OctetsRead() + aParsedMessage.mFileDesLength);
         aParsedMessage.mMetadata       = &bufStart[metadataStartIndex];
```

Comment:
This pattern means that we have to be extra-careful about lifetime, because you now have a contextual object that points to underlying storage which has to outlive this instance, but which you do not own. The function documentation is clear:
```
Note that this struct will store pointers into the passed PacketBuffer, so it
     *             is essential that the PacketBuffer is not modified or freed until after the
     *             SendAccept struct is no longer needed.
```

However, this relies on having internalized the function documentation. Suggest adding warnings to the code, where these pointers are set, that the instance has to outlive the storage.

Reply:
I'll add more comments, and also my intention is to change this once we have a way to pass reference-counted buffers.

Reply:
We have reference-counted buffers right now.  What we don't have is a nice automatic refcounting class.  But you can absolutely refcount buffers.